### PR TITLE
Prevent unexpected privileges escalation

### PR DESCRIPTION
**Pull request by gillg**
## what
The current variable `input_metadata_http_put_response_hop_limit` condition, prevent to protect users of this module, to be protected against privileges escalation.
The first intent of IMDSv2 is to prevent containers beeing able to assume an EC2 instance profile. It's not a bad idea at all to prevent that. The good practice then is to use the module `cloudposse/eks-iam-role/aws` to create a kubernetes service account mapped with IAM permissions throug an OIDC IdP.

## references
* https://aws.github.io/aws-eks-best-practices/security/docs/iam/#restrict-access-to-the-instance-profile-assigned-to-the-worker-node

### CODE REVIEW

**Comment by gillg** [comment]
Does anyone can take a look ? randomly @Gowiem

**Review by max-lobur — APPROVED** [review]

**Comment by max-lobur** [comment]
Makes sense, thanks for this. I think we will change default too, but will check regression first

**Comment by gillg** [comment]
> Makes sense, thanks for this. I think we will change default too, but will check regression first

Fine to me, but be careful with the default... ^^ A lot of people will have unexpected permissions not working anymore. It's a breaking change.

**Comment by max-lobur** [comment]
Both failures unrelated, I will fix in separate PR

**Comment by xeivieni** [comment]
Any update on this PR ? I would love to see this released 🤗

**Comment by nitrocode** [comment]
@max-lobur this is odd. It seems like it's trying to revert it back to 2022 instead of using 2023.

It should be using the latest build-harness which grabs rhe current year.

https://github.com/cloudposse/build-harness/blob/c7b01d772017cc78b3902c0a7fcd54f6b6131dca/templates/README.md.gotmpl#L288

**Comment by nitrocode** [comment]
Oh I see the issue. We just need to run the auto format again on this pr since this pr was submitted in 2022. Now its 2023 and the year should be 2023 in the readme.

Please run `make pr/auto-format` locally

**Review by nitrocode — DISMISSED** [review]

File: examples/complete/main.tf [hunk 1]
```diff
@@ -117,7 +117,7 @@ module "https_sg" {
 
 module "eks_cluster" {
   source  = "cloudposse/eks-cluster/aws"
-  version = "2.2.0"
+  version = "2.6.0"
```
Comment (nitrocode) [inline examples/complete/main.tf:120]:
```suggestion
  version = "2.4.0"
```

**Review by nitrocode — APPROVED** [review]

**Comment by nitrocode** [comment]
Thank you @gillg for the contribution!

**Pull request merged**